include/EASTL/internal/red_black_tree.h

```diff
@@ -184,6 +184,22 @@ namespace eastl
 		rbtree_iterator& operator--();
 		rbtree_iterator  operator--(int);
 
+	protected:
+		template <typename Tp, typename PointerA, typename ReferenceA, typename PointerB, typename ReferenceB>
+		friend bool operator==(const rbtree_iterator<Tp, PointerA, ReferenceA>& a, 
```

Comment:
So basically they are friend forward declarations of these functions. https://github.com/electronicarts/EASTL/blob/master/include/EASTL/internal/red_black_tree.h#L675